checker.go

```diff
@@ -269,6 +304,50 @@ func (c *hasLenChecker) Check(got interface{}, args []interface{}, note func(key
 	return nil
 }
 
+// Satisfies is a Checker checking that the provided value, when used as
+// argument of the provided function, causes the function to return true.
+// The function must be of type func(T) bool where T is the type of got.
+// For instance:
+//
+//     c.Assert(os.ErrNotExist, qt.Satisfies, os.IsNotExist)
```

Comment:
How about:

     // Check that an error from os.Open satisfies os.IsNotExist.
     c.Assert(err, qt.Satisfies, os.IsNotExist)

     // Check that a floating point number is a not-a-number.
     c.Assert(f, qt.Satisfies, math.IsNaN)

?

Reply:
Nicer.